### PR TITLE
Feature/expose return types

### PR DESCRIPTION
**Pull request by sjmarshy**
Create and use a `FineResult` type. A `FineResult` is an `fp-ts` `Either` type where the `Left` is restricted to extending `string`.

This work enables 3rd parties to easier implement `fine` interfaces. The interfaces rely on the Either type heavily, and it's absence causes problems with TypeScript. Specifically, TypeScript complains about a private type being used - which is fair enough. What's more, other versions of `fp-ts` would be incompatible and seem to cause TypeScript to hang indefinitely.

This change also leaves the ability for `fine` methods to specify exactly what possible `Left`s might be returned by a function. For example, a method could return a `FineResult<'problem one' | 'problem two', IVector>` type.

### CODE REVIEW

File: src/AffineTransform/index.ts [hunk 1]
```diff
@@ -1,92 +1,91 @@
-import { right, left, Either } from 'fp-ts/lib/Either';
+import { right, left } from 'fp-ts/lib/Either';
 
 import { IAffineTransform } from './interface';
 import { SquareMatrix3D, Matrix } from '../Matrix';
 import { IVector, Vector } from '../Vector';
-import { IPosition3D as IPos3D, Position3D } from '../Position';
-import { AxisAlignedBoundingBox } from '../AxisAlignedBoundingBox';
+import { IPosition3D as IPos3D, Position3D, IPosition3D } from '../Position';
```
[the remark this answers is not in the document]
Comment (sjmarshy) [reply src/AffineTransform/index.ts]: oops, nope! I'll get it sorted

File: src/AffineTransform/index.ts [hunk 2]
```diff
@@ -1,92 +1,91 @@
...
+import {
+  AxisAlignedBoundingBox,
+} from '../AxisAlignedBoundingBox';
```
[the remark this answers is not in the document]
Comment (sjmarshy) [reply src/AffineTransform/index.ts]: looks like the formatter didn't run for some reason, I'll give it a sweep.

File: src/AffineTransform/index.ts [hunk 3]
```diff
@@ -1,92 +1,89 @@
...
 import { AxisAlignedBoundingBox } from '../AxisAlignedBoundingBox';
 import { CubeCorners } from '../Cube';
 import { Pair } from '../util/Pair';
+import { FineResult } from '../interface';
 
 export { IAffineTransform };
 
 // TODO: export this to AffineTransform.ts
 export class AffineTransform3D implements IAffineTransform {
-    /**
```
Comment (iakomus) [inline src/AffineTransform/index.ts]: There are some format inconsistencies (2-space tabs instead of 4) that are bringing more changes than intended.

Comment (sjmarshy) [reply src/AffineTransform/index.ts]: I've added a .prettierrc and formatted those with 2-spaces to have 4

**Pull request merged**